app/src/org/commcare/fragments/personalId/PersonalIdPhoneVerificationFragment.java

```diff
@@ -162,7 +169,13 @@ private boolean shouldAutoSwitchToPersonalIdAuth(OtpErrorType errorType) {
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         binding = ScreenPersonalidPhoneVerifyBinding.inflate(inflater, container, false);
-        setupInitialState();
+
+        // If onCreateView() is being called again after process death, the user was already on this
+        // screen and receivied the initial OTP. In that case, we do not want to reset the initial state.
+        if (personalIdSessionData.getOtpAttempts() == 0) {
+            setupInitialState();
+        }
```

Comment:
> If onCreateView() is being called again after process death, the user was already on this screen and receivied the initial OTP. In that case, we do not want to reset the initial state.

Think a more concrete way of knowing that we are restoring after process death can be to check `savedInstanceState!=null`. In general this is a good pattern to follow - 

````
if(savedInstanceState!=null) {
 restoreState();
}else {
   setupInitialState();
}
````